lib/matplotlib/axis.py

```diff
@@ -1661,7 +1661,14 @@ def set_ticklabels(self, ticklabels, *, minor=False, **kwargs):
         r"""
         Set the text values of the tick labels.
 
+        .. admonition:: Discouraged
```

Comment:
perhaps combine the admonition and the warning?  they are not *exactly* the same, but still quite close...

Reply:
Done.